### PR TITLE
fix(curriculum): misleading hint on Step 28 of Cat Photo App

### LINKED ISSUE #45973: Misleading hint message after failing the test
### Describe the Issue

I nested an `img` element withing the `figure` element, then set the wrong link to the `src` attribute and checked the code but it hints that I should have a third `img` element nested in the `figure` element.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-28

### Your code

```
<figure>
  <img src="https://cdn.freecodecamp.org/curriculum/cat-photo-app/lasagna.jpg">
</figure>
```

### Expected behavior

I excpected the hint to tell me that I typed a wrong `src` value.

### Screenshots

issue

### System

- Device:Laptop
- OS: Ubuntu 20.04
- Browser: Chrome
- Version: 101.0.4951.64

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benjaminJohnson2204**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #45973

The website was giving a hint message to add an img element, even if there was an image element with an incorrect src link. With these changes, it will tell users to fix the src link if it has the incorrect link, instead of saying to add an img element.

### CODE REVIEW

**Comment by Sboonny** [comment]
Isn't with that change, what's in `src` won't matter? This should be against the instructions

**Comment by benjaminJohnson2204** [comment]
Good point. I added a new commit to still require a src element in the image.

Sorry about that, I'll focus on issues with the help wanted label from now on!

**Comment by jeremylt** [comment]
Yeah, the fix here should be to update the hint text, not the test itself.

**Comment by benjaminJohnson2204** [comment]
@jeremylt  Should one of the hints & tests just be deleted then? Before I made any code changes, there were 2 hint tests with the exact same code (they both checked that the src URL was correct).

**Comment by jeremylt** [comment]
No. All of the tests are needed. The hint text is just confusing for that particular test and needs to be reworded.

**Comment by Sboonny** [comment]
(optional) I wonder if would be more benefitable to make these two tests more lenient

```
assert(document.querySelectorAll('figure').length === 2)
assert(code.match(/<\/figure>/g).length === 2);
```
by switching `===` with `>=` there are nothing that says there should only be 2 `figures` in the test, and this shouldn't affect the test in general. because this `const catsImg = document.querySelectorAll('figure > img')[1];` will always test the 2nd one

**Comment by Sboonny** [comment]
oh btw what jeremyIt meant with merging the two

> I'm not sure how we would identify which is the third one without the src. Perhaps those two need to be merged

 these two become one

`You should have a third img element nested in the figure element.`

```
const catsImg = document.querySelectorAll('figure > img')[1];
assert(
  catsImg
);
```

`The third image should have an src attribute set to https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg.`

```
const catsImg = document.querySelectorAll('figure > img')[1];
assert(
  catsImg &&
    catsImg.getAttribute('src').toLowerCase() === 'https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg'
);
```

so it will become something like: 

`You should have a third img element nested in the figure element, with an src attribute set to https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg.`

```
const catsImg = document.querySelectorAll('figure > img')[1];
assert(
  catsImg &&
    catsImg.getAttribute('src').toLowerCase() === 'https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg'
);
```

---

but this isn't the ***important*** part, notice he said he isn't sure. so feel free to share your thoughts and how the test can be done better, that would be more than welcomed. 

if you are more comfortable with chat application, join us in [freecodecamp chat](https://chat.freecodecamp.org/channel/contributors)

**Comment by benjaminJohnson2204** [comment]
That makes sense, thanks for clarifying!

Okay for sure. I think it would be best to give as specific hints as possible. 

First, check that there is a third img. If not, tell the user 
`You should have a third img element nested in the figure element.` 

Next, check that the image has a src tag. If not, tell the user something like 
`The third image should have a src attribute.`

Finally, check that the src element is the correct link, and if not, tell the user 
`The third image should have an src attribute set to https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg.`

Thoughts @jeremylt @Sboonny ?

**Comment by Sboonny** [comment]
>Next, check that the image has a src tag. If not, tell the user something like
`The third image should have a src attribute.`
Finally, check that the src element is the correct link, and if not, tell the user
`The third image should have an src attribute set to https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg.`

I like it, it's perfect clear, because if someone typed `src` as `stc` they will get specific hint. 

```html
<figure>
  <img src="https://cdn.freecodecamp.org/curriculum/cat-photo-app/lasagna.jpg">
</figure>
```

how would you improve the test blow, if someone typed the code above?

```js
const catsImg = document.querySelectorAll('figure > img')[1];
assert(
  catsImg
);
```

**Comment by jeremylt** [comment]
I suspect the volume of conversation here is exactly why this wasn't marked as ready for contributors yet. The discussion about *potential* fixes really belongs in the issue.

**Comment by benjaminJohnson2204** [comment]
Sorry about that

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efada803cbd2bbdab94e332.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ You should have a third `img` element nested in the `figure` element.
 const catsImg = document.querySelectorAll('figure > img')[1];
 assert(
   catsImg &&
-    catsImg.getAttribute('src').toLowerCase() === 'https://cdn.freecodecamp.org/curriculum/cat-photo-app/cats.jpg'
+    catsImg.getAttribute('src')
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efada803cbd2bbdab94e332.md:38]: I would just check it like this:
`assert(catsImg)`

That pretty much just checks for exactly what the hint says. The hint doesn't say the `src` is required, so you could add `` and it wouldn't pass, even though you added the element.

**Review by moT01 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efada803cbd2bbdab94e332.md (hunk 1 above)
Comment (benjaminJohnson2204) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5efada803cbd2bbdab94e332.md]: Thanks for reviewing my pr! I just changed the code to only use `assert(catsImg)`

**Review by benjaminJohnson2204 — COMMENTED** [review]
Added requested changes to test

**Review by benjaminJohnson2204 — COMMENTED** [review]
Reviewing changes

**Comment by Sboonny** [comment]
it's normal here to have issue linger, because of the heavy load on the staff.

you can mention this PR in [freecodecamp chat](https://chat.freecodecamp.org/channel/contributors) but I wouldn't ping anyone 🤍

**Review by huyenltnguyen — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]
LGTM @benjaminJohnson2204 🎉 Thanks for contributing to freeCodeCamp 🎉

**Pull request merged**